### ISSUE TITLE
Warning about expo-analytics-segment 11.2.1 being incompatible with Expo 44 when it should work

### ISSUE THREAD

**Issue by Jorundur** [expo/expo#19020]
### Summary

There was an issue with Google Play console starting to reject applications using `expo-analytics-segment`. This was discussed here https://github.com/expo/expo/issues/17845 and fixed in version 11.2.1 as mentioned in that issue. This should work with Expo 44 as mentioned in the issue as well.

However, running `expo install expo-analytics-segment` when on Expo 44 installs version 11.1.2, not 11.2.1. When manually installing 11.2.1 instead, running `expo start` gives warning 

```
Some dependencies are incompatible with the installed expo package version:
 - expo-analytics-segment - expected version: ~11.1.2 - actual version installed: 11.2.1
Your project may not work correctly until you install the correct versions of the packages.
```

Expected behaviour: No warning when using 11.2.1 in Expo 44 as it should work.

### What platform(s) does this occur on?

Not Applicable

### Environment

```
  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 12.0.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 16.16.0 - ~/.nvm/versions/node/v16.16.0/bin/node
      Yarn: 1.22.19 - /usr/local/bin/yarn
      npm: 8.11.0 - ~/.nvm/versions/node/v16.16.0/bin/npm
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    IDEs:
      Xcode: 13.4.1/13F100 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^44.0.0 => 44.0.6
      react: 17.0.1 => 17.0.1
      react-native: 0.64.3 => 0.64.3
    npmGlobalPackages:
      eas-cli: 2.1.0
      expo-cli: 6.0.5
    Expo Workflow: managed
```

### Minimal reproducible example

1. `npx create-expo-app my-app`
2. `expo upgrade 44`
3. `expo install expo-analytics-segment`
    - Notice that this installs 11.1.2 which is incorrect
4. Manually change version in `package.json` to `~11.2.1`
5. `npm install`
6. `expo start`
    - See warning

**Comment by Jorundur** [comment]
Maybe this isn't actually an issue.

I notice now that when I run `expo install expo-analytics-segment` on Expo 44 it gets version `11.1.2` like I mentioned above and when I look into the node_module, that includes the fix for the Google Play store error (the bump to `com.segment.analytics.android:analytics:4.10.4`). So I should just be good to go with `11.1.2`.

What confused me was that nowhere did I see mentioned explicitly that version `11.1.2` fixed this, only `11.2.1` so I thought `11.2.1` was the only one that fixed it. The changelog in the `expo-analytics-segment` repo also didn't state this.

Perhaps @brentvatne can confirm and then just close this issue if this is indeed the case.

**Comment by byCedric** [comment]
Hi @Jorundur! It seems that the `expo install expo-analytics-segment` used the versions for Expo SDK 45 instead of 44. Is it possible that you had SDK 45 installed instead of 44 when you ran the `expo install` command?

You are right that the version for SDK 44 is supposed to be `11.1.2`. 

Also keep in mind that `expo-analytics-segment` is deprecated since SDK 46. If you want to use segment, its best to use the [native segment library](https://github.com/segmentio/analytics-react-native) with a dev client and/or EAS build instead.
